sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -198,6 +200,10 @@ case class CatalogTable(
       locationUri, inputFormat, outputFormat, serde, compressed, properties))
   }
 
+  def withStats(cboStatsEnabled: Boolean): CatalogTable = {
```

Comment:
Thanks. I think the first one is better, the second one will lead to many if-else on caller sides.